### PR TITLE
chore: Adding contribution guide to repository

### PR DESCRIPTION
**Pull request by jimevanssfdc**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+$ git checkout -b <name-of-the-feature>
+```
+
+### Make your changes
+
+Modify the files, lint, format and commit your code using the following commands:
+
+```bash
+$ git add <path/to/file/to/commit>
+$ git commit
```
Comment (lizaiv77) [inline CONTRIBUTING.md:129]: git commit -m `<commit message>`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+```bash
+$ git clone git@github.com:<YOUR-USERNAME>/utam-java.git
+$ cd utam-java
+$ git remote add upstream git@github.com:salesforce/utam-java.git
+```
+
+### Create a feature branch
+
+```bash
+$ git checkout main
```
Comment (lizaiv77) [inline CONTRIBUTING.md:118]: of main? maybe dev or for external contributors it's main?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+Note that a correct submission should contain passing unit tests for the modified code.
+
+### Rebase
+
+Sometimes your feature branch will get stale with respect to the master branch, and it will require a rebase. The
+following steps can help:
+
+```bash
+$ git checkout main
```
Comment (lizaiv77) [inline CONTRIBUTING.md:159]: git fetch upstream should be first here, since we're assuming user works from fork

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+$ git pull origin main
+$ git checkout <name-of-the-feature>
+$ git rebase upstream/main
```
Comment (lizaiv77) [inline CONTRIBUTING.md:162]: rebase just main, we already pulled update

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+
+Types: build, chore, ci, docs, feat, fix, perf, proposal, refactor, release, revert, style, test, wip.
+Scope: The scope should be the name of the documentation section affected (guide, tutorial, home, etc.)
+```
+
+### Update the pull request
+
+```bash
+$ git fetch origin
+$ git rebase origin/${base_branch}
```
Comment (lizaiv77) [inline CONTRIBUTING.md:188]: update usually happens against upstream i think, not origin. i might be wrong

File: CONTRIBUTING.md (hunk 1 above)
Comment (jimevanssfdc) [reply CONTRIBUTING.md:129]: `git commit` without the `-m` flag will open a text editor to allow them to add the message. Git won't let them use an empty commit message, so I'm okay with omitting the `-m` flag here.

File: CONTRIBUTING.md (hunk 3 above)
Comment (jimevanssfdc) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 5 above)
Comment (jimevanssfdc) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 2 above)
Comment (jimevanssfdc) [reply CONTRIBUTING.md]: fixed

**Review by lizaiv77 — APPROVED** [review]
LGTM

**Pull request merged**